tool/RELEASE_INSTRUCTIONS.md

```diff
@@ -168,3 +173,77 @@ have unreleased changes, publish these packages to pub.
 [README.md](https://github.com/flutter/devtools/blob/master/packages/devtools_app/release_notes/README.md)
 to add DevTools release notes to Flutter website and test them in DevTools.
 2. Once release notes are submitted to the Flutter website, send an announcement to g/flutter-internal-announce with a link to the new release notes.
+
+## Cherry-pick releases
+
+### Prepare the release in the `flutter/devtools` repo
+
+Find the [DevTools tag](https://github.com/flutter/devtools/tags) that you want to perform the cherry-pick release on top of.
+Then checkout that tag locally. For this example, we'll use `v2.29.0` as the base branch and `2.29.1` as the cherry-pick branch.
+
+```
+git checkout v2.29.0
+```
+
+Once checked out, create a new branch for your cherry picks. 
+
+```
+git checkout -b 2.29.1
+```
+
+Cherry pick the commit(s) you want in this cherry-pick release, and bump the DevTools version number:
+```
+git cherry-pick <commit>
+devtools_tool update-version auto -t patch
+```
+
+Commit your changes and push to the `upstream` remote.
+
+```
+git add .
+git commit -m "Prepare cherry-pick release - DevTools 2.29.1"
+git push upstream 2.29.1
+```
+
+Once you are completely satisfied with your changes, create a tag for this cherry-pick release:
+```
+tool/tag_version.sh
+```
+
+To move on to the next step, you will need to take note of two values:
+1) The name of the DevTools tag you just created (e.g. `v2.29.1`)
+2) The commit hash that is at the tip of this tag (see https://github.com/flutter/devtools/tags).
+
+### Manually run the DevTools Builder
+
+Follow the instructions at [go/dart-engprod/devtools.md#cherry-picks](go/dart-engprod/devtools.md#cherry-picks)
+to trigger the DevTools builder.
+
+### Create the cherry-pick CL in the Dart SDK
+
+Checkout the Dart SDK branch you want to perform the cherry-pick on top of (e.g. `stable` or `beta`),
+and create a new branch:
+
+```
+git new-branch --upstream origin/<stable or beta> cherry-pick-devtools
+```
+
+Edit the "devtools_rev" entry in the Dart SDK [DEPS](https://github.com/dart-lang/sdk/blob/main/DEPS#L104) file
+to point to the cherry-pick release hash (the commit at the tip of the cherry-pick tag you created above).
+
+Commit your changes and upload your CL:
+```
+git add .
+git commit -m "Cherry-pick DevTools 2.29.1"
+git cl upload -s
+```
+
+### Create the cherry-pick issue in the Dart SDK
+
+Follow the [Request cherry-pick approval](https://github.com/dart-lang/sdk/wiki/Cherry-picks-to-a-release-channel#request-cherry-pick-approval) instructions to create a cherry-pick request against the Dart SDK.
+
```

Comment:
Would also make sense to include instructions for merging the cherry-pick branch back to main to ensure it's commits don't get gc'ed

Reply:
I'm not sure how to do this. Let's sync on this tomorrow for the cherry pick release we just merged today, and then I will update these instructions for what that process looks like.

Reply:
Sorry, I forgot we actually have documentation for this:
https://github.com/dart-lang/sdk/wiki/Cherry-picks-to-a-release-channel#cherry-picking-a-commit-in-a-dependency

Hope that helps! Otherwise, ping me!

Reply:
@athomas I updated the instructions, though I'm still working on who has admin access to the repo settings to allow merge commits. 

@Hixie @godofredoc @devoncarew can you help? We need to land https://github.com/flutter/devtools/pull/6699 as a merge commit (not as a squash and merge). 

Would be great to have a list of a few people who have admin access to this repo that we can reach out to for temporarily adjusting repo settings for cherry-pick releases (i'd like to add this list here in the docs).

Reply:
I do have permissions, through:

> Now create a PR for this merge, and make sure to "merge" instead of "squash" it (you might have to temporarily change repo settings to do this).

this seems like a step where it's easy to make mistakes. I'm less familiar w/ the cherry pick process, but couldn't we land as a regular CL / commit to main, and then cherry-pick that to the branch channels? For fyi, engprod people w/ the right repo access are @athomas, @whesse, and @sortie

Reply:
@devoncarew Not following what you're suggesting here. This isn't about doing something in the Dart SDK repo. This has to be done in the repo of the dependency (in this case flutter/devtools).

Reply:
Sorry, I had misunderstood. I thought the merge commit was referring to the sdk repo. But from re-reading it sounds like the merge commit dance needs to happen in the flutter/devtools repo, so that that commit doesn't later get garbage collected, and we can continue to refer to it from the sdk repo.